From 627dee659dccdb3f2e30f0454c7f73caf00705f0 Mon Sep 17 00:00:00 2001
From: Doug Goldstein <cardoe@cardoe.com>
Date: Mon, 24 Nov 2025 15:04:57 -0600
Subject: [PATCH] fix: use the correct path to the image when deep image
 inspection is off

When deep image inspection is disabled, the incorrect path was used to
determine the image format of the file resulting in a no such file or
directory exception which bubbled up and made it appear as if the cache
was missing.

Change-Id: Ibaf1486da9510fdad479523159797815e783e5f6
Signed-off-by: Doug Goldstein <cardoe@cardoe.com>
---
 ironic/common/images.py                                    | 7 ++++---
 ...bug-disable-deep-image-inspection-bfd44bb8307dea1a.yaml | 7 +++++++
 2 files changed, 11 insertions(+), 3 deletions(-)
 create mode 100644 releasenotes/notes/bug-disable-deep-image-inspection-bfd44bb8307dea1a.yaml

diff --git a/ironic/common/images.py b/ironic/common/images.py
index 7da87828f..60f4a3450 100644
--- a/ironic/common/images.py
+++ b/ironic/common/images.py
@@ -527,12 +527,13 @@ def image_to_raw(image_href, path, path_tmp):
                            'format': fmt})
                 raise exception.InvalidImage()
         else:
-            fmt = get_source_format(image_href, path)
+            fmt = get_source_format(image_href, path_tmp)
             LOG.warning("Security: Image safety checking has been disabled. "
                         "This is unsafe operation. Attempting to continue "
-                        "the detected format %(img_fmt)s for %(path)s.",
+                        "with the detected format %(img_fmt)s for "
+                        "image %(image_href)s.",
                         {'img_fmt': fmt,
-                         'path': path})
+                         'image_href': image_href})

         if fmt not in RAW_IMAGE_FORMATS and fmt != "iso":
             # When the target format is NOT raw, we need to convert it.
